src/ReinforcementLearningCore/src/policies/agent/agent_base.jl

```diff
@@ -27,13 +27,31 @@ mutable struct Agent{P,T} <: AbstractPolicy
     end
 end
 
+mutable struct OfflineAgent{P,T,B,R} <: AbstractPolicy
+    policy::P
+    trajectory::T
+    behavior_agent::B #a behavior agent to fill the trajectory. Leave nothing if the trajectory is prefilled. Should share the same trajectory as the parent OfflineAgent
```

Comment:
The `behavior_` prefixes are not immediately clear / easy to grok for me, not sure if there’s another way of naming this?

Reply:
Maybe we can have a `behavior::OfflineBehavior` attribute? Added advantage would be that `OfflineAgent` would not need to be mutable, because the mutable elements could be stored in `OfflineBehavior`. Note: pretty sure `OfflineBehavior` is not a great naming idea, but you probably have a better suggestion..